src/GitTfs/Commands/Clone.cs

```diff
@@ -41,7 +41,12 @@ public OptionSet OptionSet
 
         public int Run(string tfsUrl, string tfsRepositoryPath)
         {
-            return Run(tfsUrl, tfsRepositoryPath, Path.GetFileName(tfsRepositoryPath));
+            string gitRepositoryPath;
+            if (tfsRepositoryPath == GitTfsConstants.TfsRoot)
+                gitRepositoryPath = new Uri(tfsUrl).Host;
```

Comment:
Just a question: Is the TFS host server name the best value to use? At my organization, the url looks like:

    http://our-tfs-server.our-domain.com:8080/tfs/COLLECTIONNAME/PROJECTNAME

So, with this url, `$/` refers to the "root directory" of the TFS Project Collection (i.e. `COLLECTIONNAME`). So, should we instead have `gitRepositoryPath` be the `COLLECTIONNAME` instead? There could be a situation where an organization hosts more than one TFS Project Collection on a single TFS host.

Reply:
I don't know. Maybe it should error out and prompt the user to specify a name? I think I may prefer that to guessing a name from any part of the url.

Reply:
we use a generic ``http://hostname:8080/tfs/visualstudio``, so maybe HOSTNAME_COLLECTIONNAME?

Reply:
If the user doesn't like the name, they can always rename it afterwards anyway, or just specify the name directly during the clone.

My original issue was that it errored out with "path not valid".

Reply:
I don't know what to do here...

If we prefer to get the folder name from the user, the simpler is to throw an error asking him to provide it ( which it could already do).

If you want to continue by getting the folder name from one way or another, contrary to what I first said, I'm not for getting it from the url. And this for the reason (that I just think about) the extracted name won't be short and we increase the a lot the risk to hit the 260 charcaters limit of windows path. So, if we choose this behavior finally, I prefer what you did first and clone in "repo".
And the user could rename the folder after.

Reply:
@pmiossec I don't disagree with anything you said. I think that's totally reasonable. However, it should be documented in the "man page" for this command that that's the behavior.

Reply:
@0x53A Don't forget to update`next.md` again with whatever behavior we decide to go with. 😊

Reply:
what about using ``tfs-collection``? This isn't too long, and a bit more specific than ``repo``.